graph/min_cost_b_flow/checker.cpp

```diff
@@ -0,0 +1,152 @@
+#include <vector>
+#include <utility>
+#include <exception>
+
+#include "testlib.h"
+#include "params.h"
+
+using Flow = long long int;
+using Cost = long long int;
+
+struct Edge {
+    size_t s, t;
+    Flow lower, upper;
+    Cost cost;
+};
+
+std::pair<std::vector<Flow>, std::vector<Edge>> readInstance(InStream &is) {
+    const size_t n = is.readInt();
+    const size_t m = is.readInt();
+    std::vector<Flow> bs(n);
+    std::vector<Edge> es(m);
+    for (auto &b : bs) {
+        b = (Flow) is.readLong();
+    }
+    for (auto &e : es) {
+        e.s = (size_t) is.readInt();
+        e.t = (size_t) is.readInt();
+        e.lower = (Flow) is.readLong();
+        e.upper = (Flow) is.readLong();
+        e.cost = (Cost) is.readLong();
+    }
+    return {std::move(bs), std::move(es)};
+}
+
+template<class T>
+T s2i_unchecked(std::string s, const size_t max_len) {
+    if (s.empty()) {
+        throw std::runtime_error("Got an empty string");
+    }
+    if (s.size() > max_len) {
+        throw std::runtime_error("Given string was too long: " + s);
+    }
+    bool negative = s[0] == '-';
+    if (negative) s = s.substr(1);
+    T ret{0};
+    for (const auto &c : s) {
+        if ((c < '0') || (c > '9')) {
+            throw std::runtime_error("Given string contained non-digital charactor" + s);
+        }
+        ret = (ret * 10) + (c - '0');
+    }
+    return negative ? -ret : ret;
+}
+
+template<class T>
+std::string i2s(T value) {
+    if (value < 0) return "-" + i2s(-value);
+    if (value == 0) return "0";
+    std::string s;
+    while (value) {
+        s += '0' + (value % 10);
+        value /= 10;
+    }
+    std::reverse(s.begin(), s.end());
+    return s;
+}
+
+int main(int argc, char * argv[]) {
+    registerTestlibCmd(argc, argv);
+
+    const auto instance = readInstance(inf);
+    const auto &bs = instance.first;
+    const auto &es = instance.second;
+    const size_t n = bs.size();
+    const size_t m = es.size();
+
+    const std::string first_line_of_ans = ans.readLine();
+    const std::string first_line_of_ouf = ouf.readLine();
+    if (first_line_of_ans == "infeasible") {
+        ouf.readEof();
+        if (first_line_of_ans != first_line_of_ouf) {
+            quitf(_wa, "The problem was infeasible but the first line of the output was %s", first_line_of_ouf.c_str());
+        }
+        quitf(_ok, "OK");
+    } else {
+        if (first_line_of_ans != first_line_of_ouf) {
+            quitf(_wa, "The minimum value is %s but the output was %s", first_line_of_ouf.c_str(), first_line_of_ans.c_str());
```

Comment:
I think we have a habit of making the checker check the validity of the user's output anyway and emit `_fail` for disagreement with the judge's answer. Not sure if @yosupo06 requires it for all problems though. Of course I am not doubting your solution :)

Reply:
Yes, we should use this strategy (ref: "Good checker implementation" of https://codeforces.com/blog/entry/18431) except easy problems because it is quite useful for finding the wrong of the model solution.

In this problem, I think we should use this strategy though we can use the duality to check the correctness of model solution because:

- If the model solution outputs "infeasible" for a feasible problem, we cannot detect it. (in my understanding)
- This problem isn't "easy" at least for me.

As the same with hos-lyric, I don't doubt your solution but it is better to promise correctness by many ways :)

Reply:
Thank you for the review! I love the idea and will follow that way.
-- I actually doubt my solution and that's why I've included some rationality checks on infeasible cases in the solution :)